File: src/leetcode/s001_100/leetcode_14.java
package leetcode.s001_100;

public class leetcode_14 {

    public String longestCommonPrefix(String[] strs) {
        StringBuilder s = new StringBuilder();
        if(strs.length == 0) return s.toString();
        String firstword = strs[0];
        for(int i=0;i<firstword.length();i++) {
            boolean mismatch = false;
            for(int j=1;j<strs.length;j++) {
                if(i>=strs[j].length()){
                    mismatch = true;
                    break;
                }
                if(firstword.charAt(i) != strs[j].charAt(i)) {
                    mismatch = true;
                    break;
                }
            }
            if(!mismatch) {
                s.append(firstword.charAt(i));
            } else {
                break;
            }
        }
        return s.toString();
    }

    public static void main(String[] args) {
        leetcode_14 t = new leetcode_14();
    }
}

/*
flower
flow
flight

 */

File: leetcode/s0001_0100/leetcode_14.py
from typing import *


class leetcode_14(object):
    """leetcode_14 documentation"""

    def __init__(self, *args, **kwargs):
        """Constructor for leetcode_14"""
        pass

    def solve(self):
        print(self.longestCommonPrefix(["flower","flow","flight"]))

    def longestCommonPrefix(self, strs: List[str]) -> str:
        pref = ""
        if len(strs) == 0:
            return pref
        word = strs[0]
        for i in word:
            temp = pref+i
            for j in strs[1:]:
                if j[:len(temp)] != temp:
                    return pref
            pref += i
        return pref


if __name__ == '__main__':
    l: leetcode_14 = leetcode_14()
    l.solve()
